### ISSUE TITLE
[padding-line-between-statements] Some ESLint options are not accepted by this extension rule

### ISSUE THREAD

**Issue by gebsh** [typescript-eslint/typescript-eslint#4091]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/padding-line-between-statements": [
      "warn",
      {
        "blankLine": "always",
        "prev": "cjs-export",
        "next": "*"
      }
    ]
  }
}
```

Other files are not required because this bug is related to the rule configuration, but I used an empty `.js` file when running ESLint.

**Expected Result**

ESLint does not produce any errors when validating configuration, since the `@typescript-eslint/padding-line-between-statements` rule is an extension rule, and it [should accept options provided by ESLint](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/padding-line-between-statements.md#options).

**Actual Result**

An error is thrown when running ESLint:

```text
λ ./node_modules/.bin/eslint ./test.js

Oops! Something went wrong! :(

ESLint: 8.1.0

Error: .eslintrc.js:
	Configuration for rule "@typescript-eslint/padding-line-between-statements" is invalid:
	Value "cjs-export" should be equal to one of the allowed values.
	Value "cjs-export" should be array.
	Value "cjs-export" should match some schema in anyOf.

    at ConfigValidator.validateRuleOptions (<project-root>/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:790:23)
    at <project-root>/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:845:18
    at Array.forEach (<anonymous>)
    at ConfigValidator.validateRules (<project-root>/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:842:34)
    at ConfigValidator.validateConfigArray (<project-root>/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:968:18)
    at CascadingConfigArrayFactory._finalizeConfigArray (<project-root>/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3886:23)
    at CascadingConfigArrayFactory.getConfigArrayForFile (<project-root>/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3692:21)
    at FileEnumerator._iterateFilesWithFile (<project-root>/node_modules/eslint/lib/cli-engine/file-enumerator.js:366:43)
    at FileEnumerator._iterateFiles (<project-root>/node_modules/eslint/lib/cli-engine/file-enumerator.js:347:25)
    at FileEnumerator.iterateFiles (<project-root>/node_modules/eslint/lib/cli-engine/file-enumerator.js:297:59)
```

**Additional Info**

An error is also thrown when the string `"cjs-export"` in the repro code is replaced with `"cjs-import"`. Looking at [the code](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/padding-line-between-statements.ts#L499) of this rule, it looks like options `"cjs-import"` and `"cjs-export"` were renamed to `"require"` and `"exports"` respectively, but I could not find any reasoning behind this change in [the PR](https://github.com/typescript-eslint/typescript-eslint/pull/3418) that introduced this rule. [The official ESLint documentation](https://eslint.org/docs/rules/padding-line-between-statements#rule-details) doesn't mention them too.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.3.0` |
| `@typescript-eslint/parser`        | `5.3.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `8.1.0` |
| `node`                             | `17.0.1` |

**Comment by bradzacher** [comment]
> it should accept options provided by ESLint.

There's no requirement that extension rules maintain complete compatibility with base rules. We add and remove options as required.

I'm not sure why it was changed exactly - I wasn't reviewing the original ESLint code when I reviewed the PR, as TBH complete compatibility in these stylistic extension rules is bottom of the priority list.

More than happy to accept a PR to update the docs, but it's not worth the effort to go through a breaking change to rename the options for a stylistic lint rule.

**Comment by gebsh** [comment]
> There's no requirement that extension rules maintain complete compatibility with base rules. We add and remove options as required.

Ok, thank you for the clarification.

> More than happy to accept a PR to update the docs

I will send a PR updating the docs shortly.
